src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -409,7 +420,8 @@ private void RetrieveMediaEntityStreamPaths(IEdmEntityType entityType, ODataPath
                 // ~/entityset/{key}/single-valued-Nav/subtype
                 CreateTypeCastPaths(currentPath, convertSettings, navigationProperty.DeclaringType, navigationProperty, targetsMany);
 
-                if (navigation?.Referenceable == true)
+                if (navSourceRestrictionType?.Referenceable == true ||
```

Comment:
```suggestion
                if (navSourceRestrictionType?.Referenceable ?? false ||
```

Reply:
I'm nitpicking at this point but semantically they don't mean the same thing even though they achieve the same result. ?? false goes hand in hand with the null propagation operator to say "if you don't have a value for the navSourceRestrictionType, use false as a default". Whereas == true does an implicit cast from bool? to bool

Reply:
Understood. Resolved.